src/parser/type_checking.c

```diff
@@ -5526,9 +5528,21 @@ pt_coerce_range_expr_arguments (PARSER_CONTEXT * parser, PT_NODE * expr, PT_NODE
 		  assert (false);
 		  break;
 		}
+
+	      if (PT_IS_STRING_TYPE (common_type) && PT_IS_STRING_TYPE (temp->type_enum))
+		{
+		  /* A bigger codesets's number can represent more characters. */
+		  if (units < temp->info.data_type.units)
+		    {
+		      units = temp->info.data_type.units;
+		      collation_id = temp->info.data_type.collation_id;
+		    }
```

Comment:
In the case of mixed code set, which of them should be selected ?
Currently ISO88591's code set = 3, EUCKR = 4, UTF8 = 5, therefore the selection policy you thought seems good.
The selection by code set value is dangerous a little when the code set value is changed. But it's the best in current.